in_toto/verifylib.py

```diff
@@ -240,6 +242,84 @@ def verify_layout_expiration(layout):
     raise LayoutExpiredError("Layout expired")
 
 
+def substitute_parameters(layout, parameter_dictionary):
+  """
+  <Purpose>
+    This function is a transitionary measure for parameter substitution (or
+    any other solution defined by the in-toto team). As of now, it acts as 
+    a very simple replacement layer for python-like parameters
+
+  <Arguments>
+    layout:
+            The Layout object to process.
+
+      parameter_dictionary:
+            A dictionary containing key-value pairs for substitution.
+
+  <Exceptions>
+    securesystemslib.exceptions.FormatError:
+      if the parameter dictionary is malformed.
+
+    KeyError:
+      if one of the keys in the parameter dictionary are not present for
+      substitution
+
+  <Side Effects>
+    The layout object will have any tags replaced with the corresponding
+    values defined in the parameter dictionary.
+  """
+  in_toto.formats.PARAMETER_DICTIONARY_SCHEMA.check_match(parameter_dictionary)
```

Comment:
nit: Consider including an example of a parameter_dictionary in a comment (especially since the strings being replaces are expected to contain the curly braces).  I had to refer back to the unit tests and in_toto.formats to confirm the expected format.